src/systems/systems_wasm/src/aspects.rs

```diff
@@ -0,0 +1,852 @@
+            .unwrap_or(true)
+    }
+}
+
+impl Debouncer for LeadingDebounce {
+    fn should_handle(&self) -> bool {
+        self.exceeded_debounce_duration()
+    }
+
+    fn notify_handled(&mut self) {
+        self.handled_at = Some(Instant::now());
+    }
+
+    fn post_tick(&mut self, variables: &mut MsfsVariableRegistry) {
+        if self.handled_at.is_some() && self.exceeded_debounce_duration() {
+            if let Some((variable, value)) = &self.reset_to {
+                variables.write(variable, *value);
+            }
+
+            self.handled_at = None;
+        }
+    }
+}
+
+#[derive(Clone, Copy, Default)]
+/// Configurable options for event to variable handling.
+pub struct EventToVariableOptions {
+    mask: bool,
+    leading_debounce_duration: Option<Duration>,
+    reset_to: Option<f64>,
+}
+
+impl EventToVariableOptions {
+    /// Masks the event, causing the simulator to ignore it, and only this module to receive it.
+    pub fn mask(mut self) -> Self {
+        self.mask = true;
+        self
+    }
+
+    /// Apply a leading debounce to the event handling. Leading debounce immediately executes
+    /// the action associated with the event, but ignores any subsequent event until no event
+    /// triggered for the given duration.
+    ///
+    /// This is useful to deal with poor MSFS event handling, e.g. events being triggered
+    /// repeatedly despite only one press occurring.
+    pub fn leading_debounce(mut self, duration: Duration) -> Self {
+        self.leading_debounce_duration = Some(duration);
+        self
+    }
+
+    /// Sets the value to which the variable should be reset after the event occurred and any
+    /// debounce duration has passed.
+    pub fn afterwards_reset_to(mut self, value: f64) -> Self {
+        self.reset_to = Some(value);
+        self
+    }
+}
+
+/// Declares how to map the given event to a variable value.
+pub enum EventToVariableMapping {
+    /// When the event occurs, sets the variable to the given value.
+    Value(f64),
+
+    /// Maps the event data from a u32 to an f64 without any further processing.
+    EventDataRaw,
+
+    /// Maps the event data from a 32k position to an f64.
+    EventData32kPosition,
+
+    /// When the event occurs, calls the function with event data and sets
+    /// the variable to the returned value.
+    EventDataToValue(fn(u32) -> f64),
+
+    /// When the event occurs, calls the function with the current variable value and
+    /// sets the variable to the returned value.
+    CurrentValueToValue(fn(f64) -> f64),
+
+    /// When the event occurs, calls the function with event data and the current
+    /// variable value and sets the variable to the returned value.
+    EventDataAndCurrentValueToValue(fn(u32, f64) -> f64),
+
+    /// Converts the event occurrence to a value which increases and decreases
+    /// by the given factors.
+    SmoothPress(f64, f64),
+}
+
+#[enum_dispatch]
+enum MessageHandler {
+    EventToVariable,
+}
+
```

Comment:
Are you expecting other types of `MessageHandler` in the future? Why does this need to be an enum?

Reply:
Yes. There are different types of SimConnect data coming from a message. `SimConnectRecv::Event(e)` is the one handled by `EventToVariable` now. I'd consider it likely that e.g. `SimConnectRecv::SimObjectData(data)` would later also be handled through another declarative function. There might also be other forms of event handling coming, e.g. `EventToArinc429Bus` which would connect to @beheh's work.

By adding the structure for this now, I'm making sure that any future modification will not require too much uprooting of what's already there.